common/include/pcl/common/distances.h

```diff
@@ -197,6 +197,125 @@ namespace pcl
   {
     return (std::sqrt (squaredEuclideanDistance (p1, p2)));
   }
+
+  struct PointToLineSegmentDistanceResults
+  {
+    Eigen::Vector3f points[3]; /**< \brief The points. Note: Seg = (points[1] - points[0]) */
+    Eigen::Vector3f p;         /**< \brief The closest point on line segment. */
+    float mu;                  /**< \brief The corresponding line paramentric value resulting in the minimum distance. */
+    double d;                  /**< \brief The distance between point and line segments */
+  };
+
+  /**
+    * \brief Get the distance between a point and line
+    * \param[in] p1 Origin point of the line
+    * \param[in] p2 Terminating point of the line
+    * \param[in] p The point for calulating the distance to the line
+    * \return DistPoint2LineResults
+    */
+  PCL_EXPORTS inline PointToLineSegmentDistanceResults
+  pointToLineSegmentDistance (const Eigen::Vector3f &p1, const Eigen::Vector3f &p2, const Eigen::Vector3f &p)
+  {
+    PointToLineSegmentDistanceResults results;
+
+    results.points[0] = p1;
+    results.points[1] = p2;
+    results.points[2] = p;
+
+    Eigen::Vector3f v = results.points[1] - results.points[0];
+    Eigen::Vector3f w = p - results.points[0];
+
+    double c1 = w.dot (v);
+    double c2 = v.dot (v);
+
+    if (c1 <= 0)
+      results.mu = 0.0;
+    else if (c2 <= c1)
+      results.mu = 1.0;
+    else
+      results.mu = c1 / c2;
+
+    results.p = results.points[0] + results.mu * v;
+    results.d = (p - results.p).norm ();
+
+    return results;
+  }
+
+  struct LineSegmentToLineSegmentDistanceResults
+  {
+    Eigen::Vector3f points[4]; /**< \brief The line segment points. Note: Seg1 = (points[1] - points[0]) and Seg2 = (points[3] - points[2]) */
+    Eigen::Vector3f p[2];      /**< \brief The closest point on each line segment. Note: p[0] is the closet point on Seg1. */
+    float mu[2];               /**< \brief The corresponding line paramentric value resulting in the minimum distance. */
+    double d;                  /**< \brief The distance between both line segments */
+    bool parallel;             /**< \brief Indicate that both lines are parallel. */
+  };
```

Comment:
Also I could switch to using this function lineToLineSegment possibly but may need to make a change.

Reply:
Let's keep your functions specialized for line segments which are finite. Let's make that clear in the function description, so that (contrary to me) people understand that this could mean that this distance is not necessarily measure orthogonal to the line segments.